packages/otelbin/src/contexts/EditorContext.tsx

```diff
@@ -82,13 +84,38 @@ export function useBreadcrumbs() {
 	return React.useContext(BreadcrumbsContext);
 }
 
+export function useServerSideValidationEnabled() {
+	return React.useContext(ServerSideValidationEnabled);
```

Comment:
**Question:** Why do we need a new context for this? It should be possible to always identify this from either the `useUrlState` result or `useServerSideValidation`. Contexts make it harder to maintain the code.

Reply:
@bripkens I made it as context to avoid code duplication for using useUrlState.

Reply:
If this is only about code duplication, then there are also approaches with less of a runtime impact.

```ts
export function useServerSideValidationEnabled(): Boolean {
  const [{ distro, distroVersion }] = useUrlState([distroBinding, distroVersionBinding]);
  return !!distro && !!distroVersion;
}
```

IMO context makes it harder to understand the code, because it requires more jumping around the code/understanding of the exact React tree existing at runtime.